internal/view/human_render.go

```diff
@@ -0,0 +1,248 @@
+)
+
+const defaultWidth = 78
+
+type HumanRender struct {
+	colorize *colorstring.Colorize
+	width    int
+}
+
+func NewHumanRender(disableColor bool) Render {
+	disableColor = disableColor || !term.IsTerminal(int(os.Stderr.Fd()))
+	width, _, err := term.GetSize(int(os.Stdout.Fd()))
+	if err != nil {
+		width = defaultWidth
+	}
+
+	return &HumanRender{
+		colorize: &colorstring.Colorize{
+			Colors:  colorstring.DefaultColors,
+			Disable: disableColor,
+			Reset:   true,
+		},
+		width: width,
+	}
+}
+
+func (render *HumanRender) InvalidConfigPath(filenames []string) (string, error) {
+	var buf bytes.Buffer
+
+	for _, filename := range filenames {
+		buf.WriteString(filename)
+		buf.WriteByte('\n')
+	}
+
+	return buf.String(), nil
+}
+
+func (render *HumanRender) Diagnostics(diags diagnostic.Diagnostics) (string, error) {
+	var buf bytes.Buffer
+
+	for _, diag := range diags {
+		if str := render.Diagnostic(diag); str != "" {
+			buf.WriteString(str)
+			buf.WriteByte('\n')
+		}
+	}
+
+	return buf.String(), nil
+}
+
+// Diagnostic formats a single diagnostic message.
+func (render *HumanRender) Diagnostic(diag *diagnostic.Diagnostic) string {
+	var buf bytes.Buffer
+
+	// these leftRule* variables are markers for the beginning of the lines
+	// containing the diagnostic that are intended to help sighted users
+	// better understand the information hierarchy when diagnostics appear
+	// alongside other information or alongside other diagnostics.
+	//
+	// Without this, it seems (based on folks sharing incomplete messages when
+	// asking questions, or including extra content that's not part of the
+	// diagnostic) that some readers have trouble easily identifying which
+	// text belongs to the diagnostic and which does not.
+	var leftRuleLine, leftRuleStart, leftRuleEnd string
+	var leftRuleWidth int // in visual character cells
+
+	switch hcl.DiagnosticSeverity(diag.Severity) {
+	case hcl.DiagError:
+		buf.WriteString(render.colorize.Color("[bold][red]Error: [reset]"))
+		leftRuleLine = render.colorize.Color("[red]│[reset] ")
+		leftRuleStart = render.colorize.Color("[red]╷[reset]")
+		leftRuleEnd = render.colorize.Color("[red]╵[reset]")
+		leftRuleWidth = 2
+	case hcl.DiagWarning:
+		buf.WriteString(render.colorize.Color("[bold][yellow]Warning: [reset]"))
+		leftRuleLine = render.colorize.Color("[yellow]│[reset] ")
+		leftRuleStart = render.colorize.Color("[yellow]╷[reset]")
+		leftRuleEnd = render.colorize.Color("[yellow]╵[reset]")
+		leftRuleWidth = 2
+	default:
+		// Clear out any coloring that might be applied by Terraform's UI helper,
+		// so our result is not context-sensitive.
+		buf.WriteString(render.colorize.Color("\n[reset]"))
+	}
+
+	// We don't wrap the summary, since we expect it to be terse, and since
+	// this is where we put the text of a native Go error it may not always
+	// be pure text that lends itself well to word-wrapping.
+	fmt.Fprintf(&buf, render.colorize.Color("[bold]%s[reset]\n\n"), diag.Summary)
```

Comment:
looks like error returned from `Fprintf` is not handled

Reply:
Fixed